arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/BlakeTests.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright ConsenSys Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.precompiles;
+
+import static net.consensys.linea.zktracer.module.blake2fmodexpdata.BlakeModexpDataOperation.BLAKE2f_HASH_OUTPUT_SIZE;
+import static net.consensys.linea.zktracer.module.blake2fmodexpdata.Trace.INDEX_MAX_BLAKE_RESULT;
+import static net.consensys.linea.zktracer.module.constants.GlobalConstants.LLARGE;
+
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.api.Test;
+
+public class BlakeTests {
+  @Test
+  void emptyBlakeTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(Address.BLAKE2B_F_COMPRESSION) // address
+            .push(0xffff) // gas
+            .op(OpCode.CALL)
+            .op(OpCode.POP)
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  @Test
+  void basicBlakeTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(Bytes.fromHexString("0x0badb077")) // value, some random data to hash
+            .push(5) // offset
+            .op(OpCode.MSTORE)
```

Comment:
I really don't understand  the `5` here. Is it to produce some gibberish value ? Don't get me wrong, that'll do the job, it's just surprising.

Reply:
I put a 5 just to be sure it doesn't overlap with the 4 first bytes of the `r`. But yes the value will be leftPadded in the memory, but I don't care.